### PR TITLE
[ML] Improve robustness to outliers after detecting changes in time series

### PR DESCRIPTION
**Pull request by tveasey**
We have some code which affects the weighting we apply to outliers immediately after detecting a change in a time series. This is to deal with time series which flip-flop between values: we don't want to flag every change as anomalous, but rather learn the "envelope" of the changes. However, we need to be a bit more careful than we are with allowing the model to incorporate unusual values just after a change. If we get unlucky and there is a large outlier around this time it pollutes the model.

To address the flip-flop case we simply need to increase the weight of values whose prediction error is on the same scale as the change we detected. Values whose prediction error is significantly larger can be reweighted as normal. I noticed this while debugging #2276.

**Before**
Screenshot 2022-05-24 at 12 28 55

**After**
Screenshot 2022-05-24 at 12 28 42

### CODE REVIEW

File: lib/maths/time_series/CTimeSeriesDecompositionDetail.cc [hunk 1]
```diff
@@ -317,6 +317,8 @@ const core::TPersistenceTag LAST_CHANGE_POINT_TIME_7_11_TAG{"i", "last_change_po
 const core::TPersistenceTag LAST_CANDIDATE_CHANGE_POINT_TIME_7_11_TAG{
     "j", "last_candidate_change_point_time"};
 const core::TPersistenceTag LAST_CHANGE_POINT_7_11_TAG{"k", "last_change_point"};
+// Version 8.3
+const core::TPersistenceTag WINSORIZATION_DERATE_8_3_TAG{"l", "winsorization_derate"};
```
Comment (droberts195) [inline lib/maths/time_series/CTimeSeriesDecompositionDetail.cc:321]: It's probably best to update `MODEL_SNAPSHOT_MIN_VERSION` in `CAnomalyJob.cc` to `8.3.0` to match this.

Maybe it's not that critical if it's acceptable for the data behind this tag to get lost if a job moves around in a mixed version cluster.

But bumping the version that the Java side understands will help in the long run if we ever want to remove this.

File: lib/maths/time_series/unittest/CTimeSeriesModelTest.cc [hunk 2]
```diff
@@ -294,7 +294,7 @@ void reinitializeResidualModel(double learnRate,
 
 class CDebug {
 public:
-    static const bool ENABLED{false};
+    static const bool ENABLED{true};
```
Comment (droberts195) [inline lib/maths/time_series/unittest/CTimeSeriesModelTest.cc:297]: Did you mean to leave this on for checkin?

If so, please add `results.py` to the `.gitignore` file for this directory so nobody accidentally checks it in.

File: lib/maths/time_series/CTimeSeriesDecompositionDetail.cc (hunk 1 above)
Comment (tveasey) [reply lib/maths/time_series/CTimeSeriesDecompositionDetail.cc:321]: See [3adb43b](https://github.com/elastic/ml-cpp/pull/2280/commits/3adb43bb1eed624e6e2736c4f56b54e9c5302285).

File: lib/maths/time_series/unittest/CTimeSeriesModelTest.cc (hunk 2 above)
Comment (tveasey) [reply lib/maths/time_series/unittest/CTimeSeriesModelTest.cc]: See [d0bdf29](https://github.com/elastic/ml-cpp/pull/2280/commits/d0bdf2951aebdecbd2ebc7a399cf034a5ab2bfba).

**Pull request merged**